### PR TITLE
move rag binary llmaj under rag metrics

### PR DESCRIPTION
**Pull request by lilacheden**

### CODE REVIEW

**Review by OfirArviv — APPROVED** [review]

File: prepare/metrics/llm_as_judge/binary_judge.py [hunk 1]
```diff
@@ -52,6 +51,13 @@ def get_prediction_field(metric_type):
                     infer_log_probs=use_logprobs,
                 )
 
+                add_to_catalog(
+                    metric,
+                    f"metrics.rag.{metric_type}.{inf_label}_{template_short_name}{logprobs_label}",
+                    overwrite=True,
+                )
+
+                # for backwards compatibility: keep also legacy path to metrics
```
Comment (yoavkatz) [inline prepare/metrics/llm_as_judge/binary_judge.py:60]: This is an example, where we need references and deprecation options:
https://github.com/IBM/unitxt/issues/1341

File: prepare/tasks/rag/response_generation_with_llmaj.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from typing import List, Union
```
[the remark this answers is not in the document]
Comment (yoavkatz) [reply prepare/tasks/rag/response_generation_with_llmaj.py:1]: Also then the metrics can be overridden in the load_datasets API.

[the remark this answers is not in the document]
Comment (lilacheden) [reply prepare/tasks/rag/response_generation_with_llmaj.py:1]: great, didn't know there was such option. How do I register a list of metrics as a collection?

[the remark this answers is not in the document]
Comment (lilacheden) [reply prepare/tasks/rag/response_generation_with_llmaj.py]: removed the task for now, will add the metrics list in another pr

**Review by yoavkatz — APPROVED** [review]

File: src/unitxt/api.py [hunk 3]
```diff
@@ -198,6 +198,13 @@ def infer(
         )
     predictions = post_process(raw_predictions, dataset)
     if return_data:
+        if return_meta_data:
+            infer_output_list = [
+                infer_output.__dict__ for infer_output in infer_outputs
+            ]
+            for infer_output in infer_output_list:
+                del infer_output["prediction"]
+            dataset = dataset.add_column("infer_meta_data", infer_output_list)
```
Comment (lilacheden) [inline src/unitxt/api.py:207]: @elronbandel - this part was removed in a previous commit, which breaks the llmasjudge. please allow to re-add it here.

**Pull request merged**